lib/src/main/kotlin/com/swmansion/starknet/data/ParseHex.kt

```diff
@@ -9,5 +9,10 @@ internal fun parseHex(value: String): BigInteger {
     if (!value.startsWith("0x")) {
         throw IllegalArgumentException("Hex must start with 0x")
     }
-    return BigInteger(value.removePrefix("0x"), 16)
+
+    val trimmedValue = value.removePrefix("0x").trimStart('0')
+    if (trimmedValue.isEmpty()) {
+        return BigInteger.ZERO
+    }
+    return BigInteger(trimmedValue, 16)
```

Comment:
In case of a padded hex string that represents `0`, all zeros will be trimmed. After trimming, the value will be empty - that's why there's an extra check for this edge case.

Reply:
Correct, in fact we can pass the number with padding zeros to `BigInteger` and there's no need to remove them 👍